src/Microsoft.AspNetCore.Razor.Evolution/DefaultRazorSourceDocument.cs

```diff
@@ -10,6 +10,7 @@ namespace Microsoft.AspNetCore.Razor.Evolution
     internal class DefaultRazorSourceDocument : RazorSourceDocument
     {
         private MemoryStream _stream;
+        private string _content;
 
         public DefaultRazorSourceDocument(MemoryStream stream, Encoding encoding, string filename)
```

Comment:
Yeah, this change is pretty fundamental to the rest of the changes. Keep the static factory methods around on the abstract class so it's still easy to construct one.

Just bear in mind that this will likely need to change again in the future. We can't really use strings long term because of LOH.